parquet-column/src/main/java/org/apache/parquet/io/MessageColumnIO.java

```diff
@@ -297,6 +297,11 @@ public void startMessage() {
     @Override
     public void endMessage() {
       writeNullForMissingFieldsAtCurrentLevel();
+
+      // We need to flush the cached null values before ending the record to ensure that everything is sent to the
+      // writer before the current page would be closed
+      flush();
```

Comment:
We need to run the benchmarks to see what effect this has on performance.

@isnotinvain, didn't you add some commits to delay flushing nulls to improve performance? I wonder if this reverts that change?

Reply:
Want to echo this - it looks like it effectively cancels out the null buffering from the above procedure `writeNullForMissingFieldsAtCurrentLevel`.